src/table/interfaces.tsx

```diff
@@ -406,9 +400,9 @@ export namespace TableProps {
     cancelEdit?(): void;
   }
 
-  export type SubmitEditFunction<ItemType, ValueType = any> = (
+  export type SubmitEditFunction<ItemType> = (
```

Comment:
Having an extra generic parameter allows nicer code on consumer's end

```js
const handleSubmit: TableProps.SubmitEditFunction<Item, string> = (item, column, newValue) => {
  // ... code
}
```

Which is nicer than

```js
const handleSubmit: TableProps.SubmitEditFunction<Item> = (item, column, newValue: string) => {
  // ... code
}
```

Looks odd when you annotate one argument and not others

Reply:
P.S. we might make the default generic type `unknown` which will enforce to give the annotation

Reply:
@connorlanigan suggested going with `unknown` as well, and I agree that its better than `any` in this case. I'll publish a new revision soon

Reply:
>Having an extra generic parameter allows nicer code on consumer's end

It does, but we currently can't enforce that the value we provide will match the generic parameter customers specify, see the comment thread above: https://github.com/cloudscape-design/components/pull/605#discussion_r1060726179

So `unknown` will force consumers to be aware that they have to perform either a type assertion or a type check before using it.

Reply:
This is why I proposed `type SubmitEditFunction<ItemType, ValueType= unknown> = (newValue: ValueType) => void;` and not `(newValue: unknown)` directly. Moving it into generic allows to specify the new type

[TS playground demo](https://www.typescriptlang.org/play?#code/C4TwDgpgBAygrgIwLYEtgFEAmaBicB2AxsCgPb4A8AagIYA2cEAKuBALxQEDW+pA7vgB8UDgAp8EPrQYQAXFGmMWkAJQjhAN1IpMAbgCwAKCNHC5AM7Ao5xKmABGefGRosuAsTKVLAJxT4Ac2ExCSl6RjU2YQBvIyh4qFDFCAA6YFIYYD9A0RVdKAB6Aqg+Uh8ucyMAXxNDM3xLa1s0ACYnZoxsYDwiEnIRKHFJZMiYuISk8NT0zOyA3PyiqB8IVHxMCB8oGgRSOCtUc3N-AKhQSGrapcxSCHNE0itS8qgECEIaOHNoUgAzbaOKAC+B2KDoaBAyzgdDupgsVhsLmAAGZ2ki3N0PH18AMhmEZPJfCdRlBYoYEolhlM0hksicFtUgA)

Reply:
^ @fa7ad this is the last comment/request. Can approve after this change